ush/retrieve_data.py

```diff
@@ -737,27 +756,40 @@ def main(argv):
     """
 
     cla = parse_args(argv)
+    if cla.cycle_date is None:
+        cla.cycle_date = to_datetime("1999123100") # set it to something benign
+    if cla.fcst_hrs is None:
+        cla.fcst_hrs = [0]                         # set a default
     cla.fcst_hrs = arg_list_to_range(cla.fcst_hrs)
 
     if cla.members:
         cla.members = arg_list_to_range(cla.members)
 
     setup_logging(cla.debug)
-    print("Running script retrieve_data.py with args:\n", f"{('-' * 80)}\n{('-' * 80)}")
+    print("Running script retrieve_data.py with args:", f"\n{('-' * 80)}\n{('-' * 80)}")
     for name, val in cla.__dict__.items():
         if name not in ["config"]:
             print(f"{name:>15s}: {val}")
     print(f"{('-' * 80)}\n{('-' * 80)}")
 
+    # with no --config arg passed, use our default config specs
+    if cla.config is None:
+        logging.info(f"read parm/data_locations.yml")
+        cla.config = config_exists("./parm/data_locations.yml")
+    logging.debug(f"config {cla.config}")
```

Comment:
Perhaps this, too, should be treated as a required argument. We don't want to put App-specific assumptions in the tool if we can avoid it.

Reply:
this section removed and config file arg set to required